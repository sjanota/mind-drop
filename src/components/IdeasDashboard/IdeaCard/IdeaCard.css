.IdeasDashboard .IdeaCard {
}

.IdeasDashboard .IdeaCard .card-text {
    white-space: pre-wrap;
}