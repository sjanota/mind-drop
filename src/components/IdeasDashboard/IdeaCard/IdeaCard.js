import React from 'react';
import './IdeaCard.css';
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import Octicon, {Trashcan, Pencil} from "@primer/octicons-react";
import Badge from "react-bootstrap/Badge";

function IdeaCard({card, onDelete, onEdit}) {
  return <Card style={{}}>
    <Card.Body>
      <Card.Title>{card.title}</Card.Title>
      <Card.Subtitle>{card.labels.map(label => <Badge variant={"primary"} key={label}>{label}</Badge>)}</Card.Subtitle>
      <Card.Text>{card.text}</Card.Text>
    </Card.Body>
    <Card.Footer className={"text-right"}>
      <Button variant={"secondary"} size="sm" onClick={onEdit}>
        <Octicon icon={Pencil} size='small'/>
      </Button>
      <Button variant={"danger"} size="sm" onClick={onDelete}>
        <Octicon icon={Trashcan} size='small'/>
      </Button>
    </Card.Footer>
  </Card>
}

export default IdeaCard;
