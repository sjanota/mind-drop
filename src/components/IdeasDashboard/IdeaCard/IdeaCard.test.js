import React from 'react';
import IdeaCard from './IdeaCard';
import renderer from "react-test-renderer";
import {mount} from 'enzyme';

it('matches snapshot', () => {
  const value = {
    title: "Title",
    text: "Text",
    labels: ["label1", "label2"]
  };
  const tree = renderer
    .create(<IdeaCard card={value} onEdit={jest.fn()} onDelete={jest.fn()}/>)
    .toJSON();
  expect(tree).toMatchSnapshot()
});

describe('', () => {
  const value = {
    title: "Title",
    text: "Text",
    labels: ["label1", "label2"]
  };
  const onEdit = jest.fn();
  const onDelete = jest.fn();
  const component = mount(<IdeaCard card={value} onEdit={onEdit} onDelete={onDelete}/>);

  it('calls onDelete when delete button is clicked', () => {
    component.find('button.delete').simulate('click');
    expect(onDelete).toBeCalledTimes(1);
  });

  it('calls onEdit when edit button is clicked', () => {
    component.find('button.edit').simulate('click');
    expect(onEdit).toBeCalledTimes(1);
  });
});
