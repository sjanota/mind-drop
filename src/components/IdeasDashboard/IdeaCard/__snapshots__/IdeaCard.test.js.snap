// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="IdeaCard card"
  style={Object {}}
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Title
    </div>
    <div
      className="card-subtitle h6"
    >
      <span
        className="badge badge-primary"
      >
        label1
      </span>
      <span
        className="badge badge-primary"
      >
        label2
      </span>
    </div>
    <div
      className="card-text"
    >
      <p>
        Text
      </p>
    </div>
  </div>
  <div
    className="text-right card-footer"
  >
    <button
      className="edit btn btn-secondary btn-sm"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 14 16"
        width={14}
      >
        <path
          d="M0 12v3h3l8-8-3-3-8 8zm3 2H1v-2h1v1h1v1zm10.3-9.3L12 6 9 3l1.3-1.3a.996.996 0 0 1 1.41 0l1.59 1.59c.39.39.39 1.02 0 1.41z"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <button
      className="delete btn btn-danger btn-sm"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="octicon"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "fill": "currentColor",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 12 16"
        width={12}
      >
        <path
          d="M11 2H9c0-.55-.45-1-1-1H5c-.55 0-1 .45-1 1H2c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1v9c0 .55.45 1 1 1h7c.55 0 1-.45 1-1V5c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm-1 12H3V5h1v8h1V5h1v8h1V5h1v8h1V5h1v9zm1-10H2V3h9v1z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
